bliss/simulator/prior.py

```diff
@@ -249,6 +272,45 @@ def _draw_truncated_pareto(alpha, min_x, max_x, n_samples) -> Tensor:
         uniform_samples = torch.rand(n_samples) * u_max
         return min_x / (1.0 - uniform_samples) ** (1 / alpha)
 
+    def _compute_flux_ratio(self, obj_fluxes, run, camcol, field, field_dir, gains):
+        """Query SDSS frames to get flux ratios in units of electron count.
+
+        Args:
+            obj_fluxes: tensor of electron counts for a particular SDSS object
+            run: specific scan
+            camcol: scanline within run
+            field: division of scanline
+            field_dir: directory containing frames corresponding to input field
+            gains: list containing band-specific gain values
+
+        Returns:
+            ratios (Tensor): Ratio of electron counts for each light source in field
+            relative to r-band
+        """
+        ratios = torch.zeros(obj_fluxes.size())
+
+        for i, bnd in enumerate("ugriz"):
+            # read frame corresponding to rcf and band
+            frame_path = field_dir / f"frame-{bnd}-{run:06d}-{camcol:d}-{field:04d}.fits"
+            frame = fits.open(frame_path)
+            calibration = frame[1].data  # pylint: disable=maybe-no-member
+            cal_mean = calibration.mean()
+
+            # gain varies across bands
+            bnd_gain = gains[i]
+            r_gain = gains[2]  # hard-code r-gain
+
+            # distribution of nelec_per_nmgy very clustered around mean
+            nelec_per_nmgy_bnd = bnd_gain / cal_mean
+            nelec_per_nmgy_r = r_gain / cal_mean
+            nelec_per_nmgy_rat = nelec_per_nmgy_bnd / nelec_per_nmgy_r
```

Comment:
Should this be the `r_gain` divided by the mean calibration of the r band, instead of `cal_mean`? Otherwise the division in line 306 simplifies to `bnd_gain / r_gain`.

Reply:
Yes it should, thanks for the pointer!

Reply:
I double checked, it does vary by band:
```
>>> sdss = SDSS("data/sdss", 94, 1, (12,))
>>> sdss[0]["calibration"].shape
(5, 2048)
```
So you would need to calculate the mean of the r-band calibration before the loop, and divide `r_gain` by that instead of `cal_mean`. Or if you use the SDSS object, you can do it in just two lines with `nelec_per_nmgy_list`: 
```
>>> mean_nelec_per_nmgy = sdss[0]["nelec_per_nmgy_list"].mean(axis=1)
>>> mean_nelec_per_nmgy
array([ 219.55634, 1039.936  ,  987.3069 ,  747.56915,  139.61816],
      dtype=float32)
>>> ratios = mean_nelec_per_nmgy / mean_nelec_per_nmgy[2]
>>> ratios
array([0.22237901, 1.0533057 , 1.        , 0.75718015, 0.14141314],
      dtype=float32)
```

Reply:
I just did it the latter way with the SDSS object, should be all set.